### ISSUE TITLE
gMCS Fails to Run on Low-Objective-Flux Model

### ISSUE THREAD

**Issue by Rohak72** [PlanesLab/gMCSpy#2]
Hello,

First off, thanks for creating this amazing tool -- it's been a very useful asset to my research! Everything has worked great so far but recently, when trying to run gMCSpy on a metabolic model with a low objective value (0.0006087836639733893), no solutions are yielded. This model was actually the output of a reconstruction method called E-Flux, so I'm struggling to find ways to artificially increase the objective flux without stepping on the constraints enforced by E-Flux. With that being said, I wonder if there's any way to circumvent this issue. I am more than happy to send along a copy of my model if required, along with any other inputs. Please let me know what you think, thanks!

Best,
Rohak

**Comment by Rohak72** [comment]
@cjavier98, @cjrodriguez98

**Comment by lvalcarcel** [comment]
Hi @Rohak72,

I recommend testing a change to the "targetB" parameter in the `calculateMCS `function. The "targetB" parameter represents the desired activity level of the metabolic task to be disrupted. The default value is 1e-3, but your maximum objective value is 6e-4. Therefore, I suggest adjusting "targetB" to 1e-4 or 1e-5.

If this adjustment does not yield the desired results, you can try increasing the upper and lower bounds of the model. You can do this by multiplying all lower bounds (lb) and upper bounds (ub) of the model by 100 or 1000, which should maintain the relationship between metabolic fluxes.

If these solutions don't work, please send the model to us so we can examine it more thoroughly and identify a possible solution.

Best regards,  
Luis

**Comment by Rohak72** [comment]
Hi @lvalcarcel,

Thanks for the feedback! I've tried modifying the "targetB" parameter, but it seems that only `calculateMCS()` -- not `calculateGeneMCS()` -- contains that option. Since my project aims to find candidate synthetic lethal pairs, which are expressed as sets of genes, I'd like to utilize the gMCS method if at all possible. Additionally, I attempted to multiply all of the reaction bounds (lb and ub) by the 100x and 1000x scale factors you mentioned, but that yielded no change in the results.

Do you know if there's any way to account for a "targetB"-like parameter in `calculateGeneMCS()`? If this isn't feasible yet, I completely understand! Please let me know what you think, thanks again for your help :)

Best,
Rohak

**Comment by cjrodriguez98** [comment]
Hi @Rohak72! 

I am very glad that you are using gMCSpy! 

Any feedback back is appreciated, I suggest you send me the model. (You can use https://file.io/)

And I will try the solution @lvalcarcel. If the target B solution works in calculateGeneMCS I will create and update. 

Glad to help! 

Best regards, 

Carlos

**Comment by Rohak72** [comment]
Hi @cjrodriguez98,

Thanks so much! Please find the download link to my model here: https://file.io/XAT0XWp8Esdx.

Best,
Rohak

**Comment by Rohak72** [comment]
Hi,

If you have any updates, feel free to let me know! Thanks.

Best,
Rohak

**Comment by cjrodriguez98** [comment]
Hey @Rohak72,

I've been testing the model you provided. Upon review, I found that many reactions (12,400) have had their bounds changed to be between 0 and 0 (lb=0, ub=0). These reactions should be removed.

After removing them and calculating the maximum biomass production, I found it to be 0. Therefore, I believe it is a null model, which explains why gMCSpy is unable to calculate any gMCSs, as the model is incapable of producing biomass.

Sorry for the delay.

Best regards,
Carlos
